scripts-available/CDB_CartodbfyTable.sql

```diff
@@ -649,3 +649,649 @@ BEGIN
+  RAISE DEBUG '_CDB_Rewrite_Table: %', sql;
+  EXECUTE sql;
+  
+  -- We just made a copy, so we can drop the original now
+  sql := Format('DROP TABLE %s', reloid::text);
+  RAISE DEBUG '_CDB_Rewrite_Table: %', sql;
+  EXECUTE sql;
+  
+  -- If we used a temporary destination table
+  -- we can now rename it into place
+  IF destschema = relschema THEN
+    sql := Format('ALTER TABLE %s RENAME TO %s', copyname, destname);
+    RAISE DEBUG '_CDB_Rewrite_Table: %', sql;
+    EXECUTE sql;
+  END IF;
+
+  RETURN true;
+
+END;
+$$ LANGUAGE 'plpgsql';
+
+
+
+CREATE OR REPLACE FUNCTION _CDB_Add_Indexes(reloid REGCLASS, geom_name TEXT, mercgeom_name TEXT, primary_key_name TEXT)
+  RETURNS BOOLEAN
+AS $$
+DECLARE
+  rec RECORD;
+  iname TEXT;
+  sql TEXT;
+  relname TEXT;
+BEGIN
+
+  RAISE DEBUG 'Entered _CDB_Add_Indexes';
+
+  -- Extract just the relname to use for the index names
+  SELECT c.relname
+  INTO STRICT relname
+  FROM pg_class c
+  WHERE c.oid = reloid;
+
+  -- Is the default primary key flagged as primary?
+  SELECT a.attname
+  INTO rec
+  FROM pg_class c 
+  JOIN pg_attribute a ON a.attrelid = c.oid 
+  JOIN pg_index i ON c.oid = i.indrelid AND a.attnum = ANY(i.indkey)
+  JOIN pg_class ci ON ci.oid = i.indexrelid
+  WHERE attnum > 0 
+  AND c.oid = reloid
+  AND a.attname = primary_key_name
+  AND i.indisprimary
+  AND i.indisunique
+  AND NOT attisdropped;
+  
+  -- No primary key? Add one.
+  IF NOT FOUND THEN
+    sql := Format('ALTER TABLE %s ADD PRIMARY KEY (%s)', reloid::text, primary_key_name);
+    RAISE DEBUG '_CDB_Add_Indexes: %', sql;
+    EXECUTE sql;
+  END IF;
+  
+  -- Add geometry indexes to all "special geometry columns" that 
+  -- don't have one (either have no index at all, or have a non-GIST index)
+  FOR rec IN 
+    SELECT a.attname, n.nspname
+    FROM pg_class c 
+    JOIN pg_namespace n ON n.oid = c.relnamespace
+    JOIN pg_attribute a ON a.attrelid = c.oid AND attnum > 0 
+    LEFT JOIN pg_index i ON c.oid = i.indrelid AND a.attnum = ANY(i.indkey)
+    WHERE NOT attisdropped
+    AND a.attname IN (geom_name, mercgeom_name)
+    AND c.oid = reloid
+    AND i.indexrelid IS NULL
+    UNION 
+    SELECT a.attname, n.nspname
+    FROM pg_class c 
+    JOIN pg_namespace n ON n.oid = c.relnamespace
+    JOIN pg_attribute a ON a.attrelid = c.oid AND attnum > 0 
+    JOIN pg_index i ON c.oid = i.indrelid AND a.attnum = ANY(i.indkey)
+    JOIN pg_class ci ON ci.oid = i.indexrelid
+    JOIN pg_am am ON ci.relam = am.oid
+    WHERE NOT attisdropped
+    AND a.attname IN (geom_name, mercgeom_name)
+    AND c.oid = reloid
+    AND am.amname != 'gist'
+  LOOP
+    sql := Format('CREATE INDEX %s_%s_gix ON %s USING GIST (%s)', relname, rec.attname, reloid::text, rec.attname);
+    RAISE DEBUG '_CDB_Add_Indexes: %', sql;
+    EXECUTE sql;
+  END LOOP;
+    
+  RETURN true;
+
+END;
+$$ LANGUAGE 'plpgsql';
+
+
+CREATE OR REPLACE FUNCTION CDB_CartodbfyTable2(reloid REGCLASS, destschema TEXT DEFAULT NULL)
```

Comment:
We shouldn't just think it, we should *accept* it, because whatever we think, that's what's going to happen. 3rd party importers call CDB_Cartodbfy already, lots of other CDB_* functions are already public, in that they are documented and promoted by the community team. I think marking the non-public ones, with an underbar or something, makes a lot of sense.